app/frontend/src/lib/ui/components/filterGroup.js

```diff
@@ -16,6 +16,11 @@ export const init = (groupContainerSelector, removeButtonSelector, clearButtonSe
   addFilterChangeEvent(document.getElementsByClassName(groupContainerSelector));
 
   if (document.getElementById(closeButtonSelector)) {
+    if (document.getElementById(closeButtonSelector).innerText === 'Close all' && document.getElementsByClassName('govuk-accordion__section--expanded').length === 0) {
+      document.getElementById(closeButtonSelector).innerText = 'Open all';
```

Comment:
seeing as you are using these open all and close all string in a few places would be better to make them constants like `const OPEN_ALL_BUTTON_TEXT = 'Open All'` or similar. then you could export these constants and import them into the unit test and use them where those strings are used and makes the test a bit tighter to the code its testing